src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -142,62 +152,68 @@ ReportServerSideStats(
 
   // Non-ensemble model
   if (stats.composing_models_stat.empty()) {
-    const uint64_t queue_avg_us = AverageDurationInUs(stats.queue_time_ns, cnt);
-    const uint64_t compute_input_avg_us =
-        AverageDurationInUs(stats.compute_input_time_ns, infer_cnt);
-    const uint64_t compute_infer_avg_us =
-        AverageDurationInUs(stats.compute_infer_time_ns, infer_cnt);
-    const uint64_t compute_output_avg_us =
-        AverageDurationInUs(stats.compute_output_time_ns, infer_cnt);
-    const uint64_t compute_avg_us =
-        compute_input_avg_us + compute_infer_avg_us + compute_output_avg_us;
-    const uint64_t cache_hit_avg_us =
+    const uint64_t queue_time_avg_us =
+        AverageDurationInUs(stats.queue_time_ns, stats.queue_count);
+    const uint64_t compute_input_time_avg_us = AverageDurationInUs(
+        stats.compute_input_time_ns, stats.compute_input_count);
+    const uint64_t compute_infer_time_avg_us = AverageDurationInUs(
+        stats.compute_infer_time_ns, stats.compute_infer_count);
+    const uint64_t compute_output_time_avg_us = AverageDurationInUs(
+        stats.compute_output_time_ns, stats.compute_output_count);
+    const uint64_t compute_time_avg_us = compute_input_time_avg_us +
+                                         compute_infer_time_avg_us +
+                                         compute_output_time_avg_us;
+    const uint64_t cache_hit_time_avg_us =
         AverageDurationInUs(stats.cache_hit_time_ns, cache_hit_cnt);
-    const uint64_t cache_miss_avg_us =
+    const uint64_t cache_miss_time_avg_us =
         AverageDurationInUs(stats.cache_miss_time_ns, cache_miss_cnt);
     const uint64_t total_compute_time_ns = stats.compute_input_time_ns +
                                            stats.compute_infer_time_ns +
                                            stats.compute_output_time_ns;
     // Get the average of cache hits and misses across successful requests
-    const uint64_t combined_cache_compute_avg_us = AverageDurationInUs(
-        stats.cache_hit_time_ns + stats.cache_miss_time_ns +
-            total_compute_time_ns,
-        cnt);
+    const uint64_t combined_cache_compute_time_avg_us =
+        cache_miss_cnt + cache_hit_cnt > 0
+            ? ((cache_miss_cnt *
+                (cache_miss_time_avg_us + compute_time_avg_us)) +
+               (cache_hit_cnt * cache_hit_time_avg_us)) /
+                  (cache_miss_cnt + cache_hit_cnt)
+            : 0;
```

Comment:
@matthewkotila do you have any example outputs after these changes?

I believe I initially went for a weighted average approach as well, but it made the output confusing since the breakdown in parentheses didn't sum together correctly, so I believe that's why we used the unweighted average instead. CC @tanmayv25 

So instead of something like this with unweighted average:
```
# Breakdown sums to total 10
Avg request latency: 10 usec (overhead 1 usec + queue 3 usec + cache hit/miss 6 usec)
```
I think the weighted average produced something like this output:
```
# Breakdown sums to 46 > 10
Avg request latency: 10 usec (overhead 1 usec + queue 3 usec + cache hit/miss 42 usec)
```

Reply:
@rmccorm4 Did we find why weighted average didn't work? If you look fundamentally, the total average should be weighted here right? There are N requests that go via cache path(taking X us on average) and M that go via compute path(taking Y us on average).
The standard average should be (NX + MY)/(N + M) right?
Currently we are already re-visiting the server side metrics capture and report. Let's try to make it robust and accurate.

Reply:
I'll take another look at how all the figures are being computed.

Reply:
@tanmayv25 I think the two calculations being used are more or less equivalent as you suggest. However, I think we're losing some precision due to rounding in the new method proposed here by using the already rounded averages and computing their weighted sums. I think I can write a small script to demonstrate this.

@matthewkotila is this calculation the source of the issue being fixed here?

Reply:
example:
```
root@RMCCORMICK:/mnt/triton/jira/perf-analyzer-check/server/qa/L0_perf_analyzer_report# python3 precision_test.py 
Params
---
  total_compute_time_ns = 632530983
  cache_miss_time_ns = 83699365
  cache_hit_time_ns = 18467566
  cache_hit_cnt = 4083
  cache_miss_cnt = 13138

  compute_time_avg_us = 47
  cache_hit_time_avg_us = 4
  cache_miss_time_avg_us = 6
---

Old raw result: 42.66290656756286
Old floored result: 42
New raw result: 41.382381975495036
New floored result: 41

---

# Summing total compute time, total cache hit time, total cache miss time and dividing by total cache hits + misses
Old cache/compute avg: 42

# Extrapolating weighted sums of each computed average
New cache/compute avg: 41
```

Reply:
Thanks Ryan. It makes sense now that both values are nearly identical. @matthewkotila  With these in mind, I think your previous idea just to emit an error is counts differ would be good! All these avg calculations would stay the same.